components/tab-panel/test/index.js

```diff
@@ -1,124 +1,151 @@
+			const gammaTab = getElementByClass( wrapper, 'gamma' );
+
+			const getAlphaViews = () => getElementsByClass( wrapper, 'alpha-view' );
+			const getBetaViews = () => getElementsByClass( wrapper, 'beta-view' );
+			const getGammaViews = () => getElementsByClass( wrapper, 'gamma-view' );
+
+			const getActiveTab = () => getElementByClass( wrapper, 'active-tab' );
+			const getActiveView = () => getElementByClass( wrapper, 'components-tab-panel__tab-content' ).firstChild.textContent;
+
+			expect( getActiveTab().innerHTML ).toBe( 'Alpha' );
+			expect( getAlphaViews() ).toHaveLength( 1 );
+			expect( getBetaViews() ).toHaveLength( 0 );
+			expect( getGammaViews() ).toHaveLength( 0 );
+			expect( getActiveView() ).toBe( 'alpha' );
+
+			elementClick( betaTab );
+
+			expect( getActiveTab().innerHTML ).toBe( 'Beta' );
+			expect( getAlphaViews() ).toHaveLength( 0 );
+			expect( getBetaViews() ).toHaveLength( 1 );
+			expect( getGammaViews() ).toHaveLength( 0 );
+			expect( getActiveView() ).toBe( 'beta' );
+
+			elementClick( betaTab );
+
+			expect( getActiveTab().innerHTML ).toBe( 'Beta' );
+			expect( getAlphaViews() ).toHaveLength( 0 );
+			expect( getBetaViews() ).toHaveLength( 1 );
+			expect( getGammaViews() ).toHaveLength( 0 );
+			expect( getActiveView() ).toBe( 'beta' );
+
+			elementClick( gammaTab );
+
+			expect( getActiveTab().innerHTML ).toBe( 'Gamma' );
+			expect( getAlphaViews() ).toHaveLength( 0 );
+			expect( getBetaViews() ).toHaveLength( 0 );
+			expect( getGammaViews() ).toHaveLength( 1 );
+			expect( getActiveView() ).toBe( 'gamma' );
+
+			elementClick( alphaTab );
+
+			expect( getActiveTab().innerHTML ).toBe( 'Alpha' );
+			expect( getAlphaViews() ).toHaveLength( 1 );
+			expect( getBetaViews() ).toHaveLength( 0 );
+			expect( getGammaViews() ).toHaveLength( 0 );
+			expect( getActiveView() ).toBe( 'alpha' );
 		} );
 	} );
 
 	it( 'should render with a tab initially selected by prop initialTabIndex', () => {
-		const wrapper = mount(
-			<TabPanel
-				className="test-panel"
-				activeClass="active-tab"
-				initialTabName="beta"
-				tabs={
-					[
-						{
-							name: 'alpha',
-							title: 'Alpha',
-							className: 'alpha',
-						},
-						{
-							name: 'beta',
-							title: 'Beta',
-							className: 'beta',
-						},
-					]
-				}
-			>
+		const props = {
+			className: 'test-panel',
+			activeClass: 'active-tab',
+			initialTabName: 'beta',
+			tabs: [
+				{
+					name: 'alpha',
+					title: 'Alpha',
+					className: 'alpha',
+				},
 				{
-					( tabName ) => {
-						return <p tabIndex="0" className={ tabName + '-view' }>{ tabName }</p>;
-					}
-				}
-			</TabPanel>
+					name: 'beta',
+					title: 'Beta',
+					className: 'beta',
+				},
+			],
+			children: ( tabName ) => {
+				return <p tabIndex="0" className={ tabName + '-view' }>{ tabName }</p>;
+			},
+		};
+		const wrapper = TestUtils.renderIntoDocument(
+			getTestComponent( TabPanel, props )
```

Comment:
I personally don't like the fact that we need to provide `Component` and `props` without using JSX. I hope we can improve it with proper abstraction.